public/app/plugins/datasource/influxdb/specs/query_part.jest.ts

```diff
@@ -40,5 +40,129 @@ describe('InfluxQueryPart', () => {
       expect(part.text).toBe('alias(test)');
       expect(part.render('mean(value)')).toBe('mean(value) AS "test"');
     });
+
+    it('should nest distinct when count is selected', () => {
+      var selectParts = [
+        queryPart.create({
+          type: 'field',
+          category: queryPart.getCategories().Fields,
+        }),
+        queryPart.create({
+          type: 'count',
+          category: queryPart.getCategories().Aggregations,
+        })
+      ];
+      var partModel = queryPart.create({
+        type: 'distinct',
+        category: queryPart.getCategories().Aggregations,
+      });
+
+      queryPart.replaceAggregationAdd(selectParts, partModel);
+
+      expect(selectParts[1].text).toBe("distinct()");
+      expect(selectParts[2].text).toBe("count()");
+    });
+
+    it('should convert to count distinct when distinct is selected and count added', () => {
+      var selectParts = [
+        queryPart.create({
+          type: 'field',
+          category: queryPart.getCategories().Fields,
+        }),
+        queryPart.create({
+          type: 'distinct',
+          category: queryPart.getCategories().Aggregations,
+        }),
+      ];
+      var partModel = queryPart.create({
+        type: 'count',
+        category: queryPart.getCategories().Aggregations,
+      });
+
+      queryPart.replaceAggregationAdd(selectParts, partModel);
+
+      expect(selectParts[1].text).toBe("distinct()");
+      expect(selectParts[2].text).toBe("count()");
+    });
+
+    it('should replace count distinct if an aggregation is selected', () => {
```

Comment:
There's some tests lacking when distinct is selected and you add other aggregation than count, like integral, median, mean, mode and sum. Basically if you have distinct selected and selecting for example sum the distinct part should be removed. This test verifies this for the sum aggregation and currently fails.

```javascript
it('should remove distinct when sum aggregation is selected', () => {
  var selectParts = [
    queryPart.create({
      type: 'field',
      category: queryPart.getCategories().Fields,
    }),
    queryPart.create({
      type: 'distinct',
      category: queryPart.getCategories().Aggregations,
    }),
  ];
  var partModel = queryPart.create({
    type: 'sum',
    category: queryPart.getCategories().Aggregations,
  });
  queryPart.replaceAggregationAdd(selectParts, partModel);

  expect(selectParts[1].text).toBe('sum()');
});
```

Reply:
This is now fixed and pushed